Fix #5701 with SugarTheme Enhancement

matthewpoer opened the pull request:
Custom CSS generated for the SuiteP theme was being ignored due to the subTheme system. This enhancement resolves by self-referencing for the current "primary" theme, then descending to get any files for the subTheme.

## Description
Existing customizations from prior versions of Sugar/Suite are using this architecture and it should continue to be supported, while also supporting extra CSS for the Dawn/Dark/etc. subThemes.

Issue logged as https://github.com/salesagility/SuiteCRM/issues/5701

## How To Test This
Install a custom CSS file in `custom/themes/SuiteP/css/style.css` that does something obvious, e.g. 

~~~
h2.module-title-text {
  font-weight: bold;
  color: pink;
}
~~~

QRR and notice that the change isn't seen. Apply the patch and the change will appear because this file is found and used.

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

isleshocky77 committed:
Adding the abily to relate a Note with a Campaign

isleshocky77 committed:
PSR-2 Convert tabs to spaces

isleshocky77 committed:
Code Format after tab -> spaces conversion

isleshocky77 committed:
Remove duplicate array key (clean warning)

matthewpoer committed:
Fix #5701 with SugarTheme Enhancement

Custom CSS generated for the SuiteP theme was being ignored due to the subTheme system. This enhancement resolves by self-referencing for the current "primary" theme, then descending to get any files for the subTheme.

gymad commented on include/SugarTheme/SugarTheme.php:
```diff
@@ -965,12 +965,21 @@ protected function _getImageFileName(
      * @return string path of css file to include
      */
-    public function getCSSURL($cssFileName, $returnURL = true)
+    public function getCSSURL($cssFileName, $returnURL = true, $skipSubTheme = FALSE)
```
I can not find any usages of third parameter `$skipSubTheme`. Is this function still works properly if I call it and pass `true` as `$skipSubTheme` value? Can I Test is some way? I think this function needs a unit and/or acceptance test to see it works properly in each calling situation.
Also: `FALSE` should be lowercase as it is in PSR standards what we use: 
https://www.php-fig.org/psr/psr-2/#25-keywords-and-truefalsenull

gymad replied:
@samus-aran Sorry, I was not specific enough. I meant I can not find a single call in the codebase something like this:
```
$theme->getCSSURL($cssFile, $retUrl, $thirdParameterHere); // "usages of third parameter"
```

Dillon-Brown committed:
Remove fullcalendar

Dillon-Brown committed:
Fixed #3057 - FullCalendar update to version 3.1

Dillon-Brown committed:
Fixed #4511 - Upgrade of TinyMCE to latest 4.7.x version

Dillon-Brown committed:
Remove tinyMCE

Dillon-Brown committed:
Fixed #3100 - Upgrade for HTML Purifier [Request]

Dillon-Brown committed:
Merge pull request #6714 from gymad/develop_elasticpager_fix

fix last page when modulus is zero

Dillon-Brown committed:
Merge branch 'develop' into tiny_mce

Dillon-Brown committed:
Merge pull request #6730 from Dillon-Brown/issue-6720

Fixed #6720 - Remove old recaptcha files

Dillon-Brown committed:
Merge branch 'develop' into fullcalendar

Dillon-Brown committed:
Merge branch 'develop' into fullcalendar

isleshocky77 committed:
Cleaning up the clean cache command

* Update command to allow for cache directory specified in config
* Remove warnings/errors when no cached directory present
* Update to have namespace "cache" so we can add other cache commands later
  * Leave alias for current command
* Use provided toolsets required by robo for implementation
* Print cleaned directories
* Confirm before cleaning directory
  * Allow for force override

Dillon-Brown committed:
Merge branch 'develop' into fix-clean-cache-command-config-cache-dir

isleshocky77 committed:
Code formatting

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/campaign-notes

# Conflicts:
#	modules/Campaigns/Campaign.php
#	modules/Campaigns/views/view.detail.php
#	modules/Notes/vardefs.php

Dillon-Brown committed:
Reformat

Dillon-Brown committed:
Fix test

samus-aran committed:
Merge pull request #4606 from isleshocky77/feature/campaign-notes

Add ability to have Notes on a Campaign

Dillon-Brown committed:
Formatting

Dillon-Brown committed:
Merge pull request #6802 from isleshocky77/fix-clean-cache-command-config-cache-dir

Allow for clean cache command to use config'd cache_dir (plus clean up)

Dillon-Brown committed:
Remove PHP4 constructors

Dillon-Brown committed:
Nusoap PHP4 style constructors

Dillon-Brown committed:
More PHP4 style constructors

Dillon-Brown committed:
Review comments

cherrador committed:
Update modules/AOR_Charts/lib/pChart/class/pRadar.class.php

Co-Authored-By: Dillon-Brown <26431166+Dillon-Brown@users.noreply.github.com>

lazka committed:
Fixes #7111 - composer: include mpdf/mpdf 6.1 and drop in-tree version (5.7)

Remove the included version of mpdf (in modules/AOS_PDF_Templates/PDF_Lib) and
add it to composer instead.

This also updates the version from 5.7.1 to 6.1.0 to get some fixes that were already
applied manually to the version in SuiteCRM. There are many newer versions available,
but they require more changes and this is a good first step (and this potentially
could be backported to 7.10)

The version in tree had one manual fix applied (d1279cad9c717) which is included in
6.1.0 already, so the upstream version can be used as is.

The required changes to make things compatible with with 6.1 were
* mPDF_exception is now called MpdfException
* SetAutoFont() no longer exists, the docs suggest to use autoLangToFont instead
  to get the same results as before.

Fixes #7111

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into fullcalendar

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop_merge

code-ph0y committed:
Merge pull request #6702 from Dillon-Brown/fullcalendar

Fixed #3057 - FullCalendar update to version 3.1

Dillon-Brown committed:
Merge branch 'develop' into develop_merge

samus-aran committed:
Merge pull request #6703 from Dillon-Brown/htmlpurifier

Fixed #3100 - Upgrade for HTML Purifier [Request]

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/master' into develop_merge

Dillon-Brown committed:
Merge remote-tracking branch 'origin/develop_merge' into develop_merge

# Conflicts:
#	composer.json

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge pull request #7131 from Dillon-Brown/develop_merge

Develop merge

Dillon-Brown committed:
Merge branch 'develop' into remove_constructors

Dillon-Brown committed:
Add pclzip/pclzip to composer

Dillon-Brown committed:
Remove pclzip references

Dillon-Brown committed:
Remove pclZip from include

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Remove zip_utils.php

Dillon-Brown committed:
Reformat php_zip_utils.php

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into tiny_mce

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
Update tinymce references

Dillon-Brown committed:
Add tedivm/jshrink to composer

Dillon-Brown committed:
Remove Minifier.php

Dillon-Brown committed:
Fixed #7027 - Use a maintained library to minify JS

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into fix-mpdf-composer

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
PHP 5.6 compat

code-ph0y committed:
Merge pull request #6699 from Dillon-Brown/tiny_mce

Fixed #4511 - Upgrade of TinyMCE to latest 4.9.x version

Dillon-Brown committed:
Merge branch 'develop' into fix-mpdf-composer

Dillon-Brown committed:
Merge pull request #7118 from lazka/fix-mpdf-composer

Fixes #7111 - composer: include mpdf/mpdf 6.1 and drop in-tree version (5.7)

code-ph0y committed:
Merge branch 'develop' into issue-7027

cameronblaikie committed:
Merge pull request #7295 from Dillon-Brown/issue-7027

Fixed #7027 - Use a maintained library to minify JS

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/plcZip

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/plcZip

# Conflicts:
#	composer.lock

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into remove_constructors

# Conflicts:
#	modules/AOS_PDF_Templates/PDF_Lib/classes/bmp.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/directw.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/form.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/gif.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/grad.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/indic.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/svg.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/tocontents.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/wmf.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdf.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/fpdi_pdf_parser.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_context.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_parser.php

Dillon-Brown committed:
Merge pull request #7287 from Dillon-Brown/feature/plcZip

Remove PclZip

connorshea committed:
Remove HTMLPurifier ConfigForm.js file.

This probably isn't necessary and can be removed?

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop_merge_1

# Conflicts:
#	composer.lock
#	files.md5
#	include/HTMLPurifier/HTMLPurifier.standalone.php
#	include/HTMLPurifier/standalone/HTMLPurifier/Lexer/PH5P.php
#	include/SugarCharts/Jit/FlashCanvas/canvas2png.js
#	include/SugarCharts/Jit/FlashCanvas/flashcanvas.js
#	include/SugarCharts/Jit/js/Jit/jit.js
#	include/SugarCharts/Jit/js/sugarCharts.js
#	include/pclzip/pclzip.lib.php
#	include/utils/zip_utils.php
#	lib/Robo/Plugin/Commands/CleanCacheCommands.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/barcode.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/bmp.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/directw.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/form.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/gif.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/grad.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/indic.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/meter.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/svg.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/tocontents.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni_analysis.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/wmf.php
#	modules/AOS_PDF_Templates/PDF_Lib/compress.php
#	modules/AOS_PDF_Templates/PDF_Lib/config.php
#	modules/AOS_PDF_Templates/PDF_Lib/config_cp.php
#	modules/AOS_PDF_Templates/PDF_Lib/config_fonts.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourier.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourierb.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourierbi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourieri.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelvetica.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelveticab.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelveticabi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelveticai.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/csymbol.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimes.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimesb.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimesbi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimesi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/czapfdingbats.php
#	modules/AOS_PDF_Templates/PDF_Lib/graph.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/CJKdata.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/functions.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_bn_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_gu_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_hi_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_kn_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_ml_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_or_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_pa_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_ta_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_te_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/out.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/subs_core.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/subs_win-1252.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/upperCase.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdf.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/filters/FilterASCII85.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/filters/FilterLZW.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/fpdi_pdf_parser.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_context.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_parser.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/de.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/en.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/es.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/fi.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/fr.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/it.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/nl.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/pl.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/sv.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/UnicodeRanges.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_collections.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_coverage.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_dump.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_names.php
#	modules/Emails/include/ListView/DeleteEmailAction.js
#	modules/UpgradeWizard/silentUpgrade_step1.php

Dillon-Brown committed:
Update composer.lock

Mac-Rae commented:
Please resolve the conflicts on this pull request bringing it up to date.

samus-aran committed:
Merge pull request #7674 from Dillon-Brown/develop_merge_1

Develop merge

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into remove_constructors

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into remove_constructors

# Conflicts:
#	include/nusoap/class.soap_parser.php
#	include/nusoap/class.xmlschema.php

HamburgischerVereinSeefahrt committed:
The current SuiteP Theme does currently not allow to differentiate between different modules in the html code. While this is not necessary for the standard SuiteP layout, it would allow for tiny customization with css here and there. The present commit adds a "data-module" tag to the SuiteP pagecontent div and "data-field" tags in the edit and detail views. This allows developers to easily customize the appearance of individual views by simply adding custom css code, e.g.:
div#pagecontent[data-module='Accounts'] .edit-view-row-item[data-field='assigned_user_name'] {color: red;}

samus-aran committed:
Merge pull request #6750 from Dillon-Brown/remove_constructors

Remove PHP4 style constructors

Dillon-Brown committed:
Merge pull request #7606 from connorshea/remove-orphaned-html-purifier-file

Remove HTMLPurifier ConfigForm.js file.

connorshea committed:
Remove dead code.

As far as I've been able to tell, this code is entirely unused. So I've removed it.

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop_merge_4

# Conflicts:
#	codeception.yml
#	composer.json
#	composer.lock
#	include/HTMLPurifier/HTMLPurifier.standalone.php
#	include/HTMLPurifier/standalone/HTMLPurifier/Printer/ConfigForm.php
#	include/MVC/View/SugarView.php
#	include/nusoap/class.nusoap_base.php
#	include/nusoap/class.soap_transport_http.php
#	include/nusoap/class.soapclient.php
#	include/nusoap/class.wsdl.php
#	include/nusoap/class.wsdlcache.php
#	include/parsecsv.lib.php
#	modules/AOR_Charts/lib/pChart/class/pBarcode128.class.php
#	modules/AOR_Charts/lib/pChart/class/pBarcode39.class.php
#	modules/AOR_Charts/lib/pChart/class/pCache.class.php
#	modules/AOR_Charts/lib/pChart/class/pData.class.php
#	modules/AOR_Charts/lib/pChart/class/pImage.class.php
#	modules/AOR_Charts/lib/pChart/class/pSpring.class.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/barcode.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/gif.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/grad.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni.php
#	modules/AOS_PDF_Templates/PDF_Lib/compress.php
#	modules/AOS_PDF_Templates/PDF_Lib/graph.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/functions.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/no_image.jpg
#	modules/AOS_PDF_Templates/PDF_Lib/mpdf.php
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_diagonals-thick_90_eeeeee_40x40.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_glass_100_e4f1fb_1x400.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_glass_50_3baae3_1x400.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_glass_80_d7ebf9_1x400.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_highlight-hard_100_f2f5f7_1x100.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_highlight-hard_70_000000_1x100.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_2694e8_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_2e83ff_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_3d80b3_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_72a7cf_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_ffffff_256x240.png
#	modules/Employees/Employee.php

Dillon-Brown committed:
Update composer.lock

connorshea committed:
Remove deprecated DBHelper class.

This hasn't been used for over 6 years. Time to clean it up.

connorshea committed:
Remove FreeTDSHelper.

Deprecated for years.

connorshea committed:
Remove MssqlHelper.

Deprecated for many years.

The method referenced in SqlsrvManager.php's comments doesn't seem to
exist anywhere, so I've removed the reference entirely.

connorshea committed:
Remove MysqlHelper.

It's been deprecated for many years now.

cameronblaikie committed:
Merge pull request #7844 from HVStechnik/feature/SuiteP-add-identifying-html-module-and-view-tag

SuiteP: Add html data tags to allow module and field identification

Dillon-Brown committed:
Merge pull request #8074 from connorshea/remove-deprecated-helper-files

Remove deprecated helper files

Dillon-Brown committed:
Merge pull request #8084 from connorshea/activity-subpanel-view-smarty

Remove dead code: Re:Remastered

connorshea committed:
Remove HTMLSax3.

This was only ever used as part of HTMLPurifier (as an experimental
parser), as far as I can tell. Since HTMLPurifier is no longer vendored
in the repository, this library no longer needs to be vendored either.

Dillon-Brown committed:
Merge pull request #8088 from connorshea/remove-pear-html-sax

Remove XML_HTMLSax3

Dillon-Brown committed:
Merge pull request #8066 from Dillon-Brown/develop_merge_4

Develop merge

connorshea committed:
Remove all deprecated Smarty files.

Smarty is included via Composer now, so these files can be removed. If
anyone is requiring these files in their custom code, they
should make sure to remove those requires.

All Smarty plugins remain as-is.

connorshea committed:
Remove SugarXHProf.

The XHProf project has been abandoned for 4 years, and
the version vendored here is much older than even that. It's
undocumented, unmaintained, and it doesn't seem like it even works
on PHP 7.x.

"The best way to choose what to keep and what to throw away
is to take each item in one’s hand and ask: 'Does this spark joy?'
If it does, keep it. If not, dispose of it."

- Marie Kondō

connorshea committed:
Remove deprecated encrypt_password function for User.

It's not used anywhere else in the codebase besides the test
suite, should be safe to remove.

connorshea committed:
Remove deprecated JSON functions.

These have been deprecated for over six years now.

There was only one use of encodeReal, so I've replaced it.

connorshea committed:
Remove unused, deprecated functions from DBManager.php.

This leaves getHelper() for now because it's still used in a
few places. As far as I can tell, all of these have been deprecated
since at  least the fork from SugarCRM.
None of the removed functions are being used elsewhere in the codebase.

connorshea committed:
Remove deprecated getReminderTime function in CallHelper.

connorshea committed:
Remove remove_redundant_http() from Account.

This code doesn't do anything and hasn't been used for a long time.

Also removes the test for it.

connorshea committed:
Remove get_day_by_index_this_year from SugarDateTime.

This has been deprecated for a long time and is unused.

Dillon-Brown committed:
Merge pull request #8128 from connorshea/remove-deprecated-sugar-date-time-function

Remove get_day_by_index_this_year from SugarDateTime.

Mac-Rae committed:
Merge pull request #8127 from connorshea/remove-remove-redundant-http-function

Remove remove_redundant_http() from Account.

Mac-Rae committed:
Merge pull request #8126 from connorshea/remove-deprecated-call-helper-function

Remove deprecated getReminderTime function in CallHelper.

Mac-Rae committed:
Merge pull request #8125 from connorshea/remove-deprecated-code-from-dbmanager

Remove unused, deprecated functions from DBManager.php.

Mac-Rae committed:
Merge pull request #8124 from connorshea/remove-deprecated-json-functions

Remove deprecated JSON functions encodeReal() and decodeReal()

Mac-Rae committed:
Merge pull request #8122 from connorshea/remove-deprecated-encrypt-function

Remove deprecated encrypt_password function for User.

Mac-Rae committed:
Merge pull request #8095 from connorshea/remove-xhprof

Remove SugarXHProf

Mac-Rae committed:
Merge pull request #8093 from connorshea/remove-deprecated-smarty

Remove all deprecated Smarty files.

connorshea committed:
Remove the deprecated load_menu() function in utils.php

It's been deprecated since SuiteCRM's initial commit and a grep
for load_menu doesn't return anything, so it doesn't seem to be used
anywhere.

connorshea committed:
Remove deprecated functions from utils.php.

Remove get_admin_modules_for_user, get_theme_display, get_themes,
and isValidId.

These are all deprecated and unused.

Dillon-Brown committed:
Merge pull request #8218 from connorshea/remove-deprecated-functions

Remove deprecated functions from utils.php.

Dillon-Brown committed:
Merge pull request #8217 from connorshea/remove-deprecated-load-menu-function

Remove the deprecated load_menu() function in utils.php

Mac-Rae committed:
Merge branch 'develop' into matthewpoer-5701
